src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php

```diff
@@ -83,6 +84,13 @@ private function recordConstDefinition(Node $node)
         }
     }
 
+    private function recordDefinedConstDefinition(Node $node)
+    {
+        if ($node instanceof Node\Expr\FuncCall && $node->name instanceof Node\Name && $node->name->toString() === 'define') {
```

Comment:
This needs to verify two things:

 1. `define()` is not namespaced (script with imported `define()` symbol call needed)
 2. `$node->args[0]->value->value` is not guaranteed to be a constant: could be an expression. In that case, this logic needs to bail out. A test using a dynamic `define()` (just a variable instead of a string constant is sufficient) is needed

Reply:
I added a test for using a dynamic value, too.
But I don't understand what you try to say with your first point and don't really get the reason why the scruitinizer code coverage dropped.

Reply:
@maglnet https://3v4l.org/YhkJY

```php
<?php

namespace foo
{
    function define($a, $b) {
        echo 'HAHA! I don\'t really define anything!';
    }
    
    define('aa', 'bb');
}
```

Reply:
Ahhhh, yes, now I got it.
I'll add a test for this soon. :+1: